haskell-indexer-plugin/src/Haskell/Indexer/Plugin.hs

```diff
@@ -0,0 +1,56 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE OverloadedStrings #-}
+module Haskell.Indexer.Plugin where
+
+import GhcPlugins hiding (($$))
+import TcRnTypes
+
+import TcRnMonad
+import TcRnDriver
+import GhcMonad
+import Data.IORef
+
+import Language.Haskell.Indexer.Backend.Ghc
+import Language.Haskell.Indexer.Backend.GhcEnv
+import Language.Haskell.Indexer.Translate
+import Language.Haskell.Indexer.Args
+
+import System.IO
+import System.FilePath
+import System.Directory
+
+plugin :: Plugin
+plugin = defaultPlugin
+          { renamedResultAction   = keepRenamedSource
+          , typeCheckResultAction = install }
+
+mkPath :: FilePath -> Module -> FilePath
+mkPath fp m
+  = fp </> (moduleNameString (moduleName m) ++ (show (moduleUnitId m)))
+
+install :: [CommandLineOption] -> ModSummary -> TcGblEnv -> TcM TcGblEnv
```

Comment:
Please comment if this is invoked per-module? (or per-whatever). Helps understanding the outpath creation behavior (or at least dismisses the need to assume).